presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java

```diff
@@ -362,13 +366,28 @@ public MaterializedResult execute(Session session, @Language("SQL") String sql)
     {
         lock.readLock().lock();
         try {
-            return prestoClient.execute(session, sql).getResult();
+            MaterializedResult result = prestoClient.execute(session, sql).getResult();
+            checkExplainSucceeds(sql, query -> prestoClient.execute(session, query));
+            return result;
         }
         finally {
             lock.readLock().unlock();
         }
     }
 
+    private void checkExplainSucceeds(String sql, Consumer<String> queryExecutor)
+    {
+        if (sql.toUpperCase(ENGLISH).contains("SKIP EXPLAIN")) {
```

Comment:
Instead of matching strings, consider adding a variant of assertQuery that disables the EXPLAIN check and adjust the call sites accordingly.